src/main/kotlin/blackjack/controller/Controller.kt

```diff
@@ -0,0 +1,116 @@
+package blackjack.controller
+
+import blackjack.model.Deck
+import blackjack.model.FinalResult
+import blackjack.model.GamblerInfo
+import blackjack.model.Player
+import blackjack.view.InputView
+import blackjack.view.OutputView
+
+class Controller() {
+    private var players: List<Player> = mutableListOf()
+    private val dealer: Player = Player(GamblerInfo("dealer"))
+    private val deck = Deck()
+
+    fun run() {
+        try {
+            createPlayers()
+            OutputView.displayNamesOfPlayers(players)
+            roundOne()
+            OutputView.displayCardsOfDealer(dealer)
+            players.forEach { OutputView.displayCardsOfPlayers(it) }
+            OutputView.printEmptyLine()
+            players.forEach { playerTakesTurn(it) }
+            dealerTakesTurn()
+            OutputView.printEmptyLine()
+            OutputView.displayCardsOfPlayersWithScore(dealer)
+            players.forEach { OutputView.displayCardsOfPlayersWithScore(it) }
+            OutputView.printEmptyLine()
+            OutputView.displayFinalResultsHeading()
+            showResults(FinalResult(dealer, players))
+        } catch (err: IllegalArgumentException) {
+            OutputView.displayErrorMessages(err.message)
+        }
+    }
+
+    private fun createPlayers() {
+        players =
+            processPlayerNames().map { Player(it) }
+    }
+
+    private fun roundOne() {
+        dealer.addCard(deck.drawCard(2))
+        players.forEach { it.addCard(deck.drawCard(2)) }
+    }
+
+    fun processPlayerNames(): List<GamblerInfo> {
+        repeat(MAX_ATTEMPTS) {
+            try {
+                val names = InputView.getNamesOfPlayers()
+                return names
+                    .split(",")
+                    .map(String::trim)
+                    .map { it -> GamblerInfo(it) } // to avoid abundant function: caller
```

Comment:
```suggestion
                    .map { GamblerInfo(it) }
```

We can remove the arrow in this kind of lambda expression :)
Please check other parts of this project as well.

Also, the above refactored logic can also be:  

```kotlin
                    .map(::GamblerInfo)
```

Reply:
Just like the `TODO` review,
I don't this comment `to avoid abundant function: caller` is necessary here.
I didn't fully understand what it's trying to say, so it takes more time to understand the
intention. Why don't we remove it, and from now on, try best to let our code explain itself 🙂 .

To go beyond, let's think about the situation where we might actually need comments, and how to write / keep them so that it doesn't get mixed with our logic.

To get some idea, it's great to look into Kotlin and other libraries' implementation.